### PR TITLE
NEO-2315  SplitButton PR for Design

### PR DESCRIPTION
**Pull request by yangAtSpoken**
[Please check both stories under SplitButton](https://deploy-preview-473--neo-react-library-storybook.netlify.app/?path=/docs/components-splitbutton--docs)

**Before tagging the team for a review, I have done the following:**

- [x] run `yarn all` locally: ensures that all tests pass, formatting is done, types pass, and builds pass
- [x] reviewed my code changes
- [x] updated the link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] done an accessibility check on my work (tested with Chrome's `axe Dev Tools`, Mac's VoiceOver, etc.)
- [x] tagged `@avaya-dux/dux-design` if any visual changes have occurred
- [x] tagged `@avaya-dux/dux-devs`

ADD PR SUMMARY (should be brief, one sentence if possible)

`@avaya-dux/dux-design`: 
- check everything please.

`@avaya-dux/dux-devs`: 
- added an option createFirstMenuItem;
- verify user provided first menu item;
- minor css update;

## Summary by CodeRabbit

- **New Features**
	- Enhanced the `SplitButton` component for improved handling of menu items, ensuring consistent behavior when the first menu item is absent.
	- Introduced a utility function to validate the first menu item of the `SplitButton`, enhancing robustness and preventing potential errors.

- **Style Updates**
	- Updated border styling and margin adjustments for better maintainability and improved visual alignment in the `SplitButton` component.
	- Added new styles for button border-radius and dropdown minimum width to enhance overall design consistency.

- **Usability Improvements**
	- Added an interactive `onClick` event handler in the storybook for the `SplitButton`, improving user engagement during testing.

### CODE REVIEW

File: src/components/SplitButton/SplitButton_shim.css [hunk 1]
```diff
@@ -10,7 +10,7 @@
 }
 
 .neo-splitbutton--secondary .neo-splitbutton__button {
-  border: 1px solid #1B77AF;
+  border: 1px solid var(--neo-color-blue-500);
```
Comment (joe-s-avaya) [inline src/components/SplitButton/SplitButton_shim.css:13]: If you haven't added dark mode colors yet you will need to do that

File: src/components/SplitButton/helper.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import { reactNodeToString } from "utils";
+
+export const verifyFirstMenuItem = (
+	buttonText?: string,
+	onClick?: () => void,
+	children?: ReactNode,
+) => {
+	if (!children || Children.count(children) === 0) {
+		throw new Error(
+			"The first menu item must be provided if createFirstMenuItem is false.",
```
Comment (joe-s-avaya) [inline src/components/SplitButton/helper.ts:12]: You mentioned `createFirstMenuItem` but didn't use it in this file. If you mention it here, you should use it here.

**Review by joe-s-avaya — APPROVED** [review]
Makes sense. I have left a recommendation for improving the maintainability, but this looks good. Also, excellent tests 👍

File: src/components/SplitButton/SplitButton_shim.css (hunk 1 above)
Comment (joe-s-avaya) [reply src/components/SplitButton/SplitButton_shim.css:13]: Actually, now that I'm thinking about it, you said that we should have a separate Jira Issue for the neo-css update. I'll create a jira issue for moving the css over and adding dark mode. It'll be way easier to test in neo-css anyway.

Comment (joe-s-avaya) [reply src/components/SplitButton/SplitButton_shim.css:13]: I've [created a ticket](https://avaya.atlassian.net/browse/NEO-2336) for this work and put it at the top of the backlog. I've set it as a two-point story. Please update as you see fit.

**Comment by matthewAvaya** [comment]
- For the secondary split buttons, the hover color for the text, icon, and border should be #155E8B. It looks like the border color for the hover state is using the same color as the pressed state. 
- The dropdown menu is quite wide. Make it less wide so there's not so much empty white space.

File: src/components/SplitButton/SplitButton_shim.css (hunk 1 above)
Comment (yangAtSpoken) [reply src/components/SplitButton/SplitButton_shim.css:13]: @matthewAvaya Do we have dark mode colors specified?

File: src/components/SplitButton/helper.ts (hunk 2 above)
Comment (yangAtSpoken) [reply src/components/SplitButton/helper.ts:12]: I do not agree with this.

**Review by matthewAvaya — APPROVED** [review]

**Pull request merged**